x-pack/plugins/security_solution/public/flyout/right/utils/analyzer_helpers.ts

```diff
@@ -4,69 +4,132 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
+import React from 'react';
 import type { Node } from '@elastic/eui/src/components/tree_view/tree_view';
+import { EuiToken } from '@elastic/eui';
 import type { StatsNode } from '../../../common/containers/alerts/use_alert_prevalence_from_process_tree';
 
 /**
- * Helper function to check if a node has grandchildren
- * @param statsNodes type StatsNode array
- * @param children type StatsNode array
- * @return a boolean of wheather a bode has grandchildren
+ * Helper function to recursively create ancestor tree nodes
+ * @param statsNodes array of all statsNodes
+ * @param node current node which we populate the parents for
+ * @param treeNode current node in tree node form
+ * @param ancestorLevel number of level of ancestor to reach
+ * @param idx current level number for comparison to ancestorLevel
+ * @return a node list for EuiTreeView
  */
-export const hasGrandchildren = (statsNodes: StatsNode[], children: StatsNode[]) => {
-  return children.some((child) => statsNodes.some((node) => node.parent === child.id));
+export const getAncestorTreeNodes = (
+  statsNodes: StatsNode[],
+  node: StatsNode,
+  treeNode: Node,
+  ancestorLevel: number,
+  idx: number
+): Node => {
+  const ancestorNode = statsNodes.find((parent) => parent.id === node.parent);
+
+  if (idx === ancestorLevel && ancestorNode) {
+    return {
+      id: `ancestor`,
+      label: '...',
+      children: [treeNode],
+      isExpanded: true,
+    };
+  }
+  if (ancestorLevel < 0 || idx < 0 || idx >= ancestorLevel || !ancestorNode) {
+    return treeNode;
+  }
+
+  const ancestorTreeNode = {
+    id: ancestorNode.id,
+    label: ancestorNode.name,
+    children: [treeNode],
+    isExpanded: true,
+  };
+  return getAncestorTreeNodes(statsNodes, ancestorNode, ancestorTreeNode, ancestorLevel, idx + 1);
 };
 
 /**
- * Helper function to create a tree node from id, name and children
- * @param id type string
- * @param name type string
- * @param children children nodes
- * @return a node of Node type
+ * Helper function to recursively create descendant tree nodes
+ * @param statsNodes array of all statsNodes
+ * @param node current node which we populate the children for
+ * @param childrenCountLimit number of children displayed, default to 3
+ * @param descendantLevel number of level of descendants to reach
+ * @param idx current level number for comparison to descendatLevel
+ * @return a node list for EuiTreeView
  */
-export const getTreeNode = (id: string, name: string, children: Node[]): Node => {
-  return { label: name, id, isExpanded: true, children };
+export const getDescendantTreeNodes = (
+  statsNodes: StatsNode[],
+  node: StatsNode,
+  childCountLimit: number,
+  descendantLevel: number,
```

Comment:
do we need both `childCountLimit` and `descendantLevel`? Couldn't we get away with only a single `descendantLevel` propery?

Reply:
@PhilippeOberti `childCountLimit` determines how many children to display per level, and `descendantLevel` determines how deep (number of level of descendants). Keeping them separate allows flexibility for quick changes if we receive feedback from product 😅

![image](https://github.com/elastic/kibana/assets/18648970/b10d4df5-5ffc-4d81-b715-d4b724d7f6e8)